File: 24-swap-nodes-in-pairs/24-swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None:
            return None
        if head.next is None:
            return head
        else:
            temp = head.next
            head.next = head.next.next
            temp.next = head
            head = temp
            
            head.next.next = self.swapPairs(head.next.next)
            return head
            

File: 24-swap-nodes-in-pairs/24-swap-nodes-in-pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode *temp;
        if(head == NULL){
            return NULL;
        }else if(head->next == NULL) {
           return head;
        }else{
            temp = head->next;
            head->next = temp->next;
            temp->next = head;
            head = temp;
        }
        head -> next ->next = swapPairs(head->next->next);
        return head;
    }
};